### PR TITLE
Adds modules for alpha equivalence, name equivalence, and exact text matching

### PR DESCRIPTION
**Pull request by dhsorens**
## Overview
1. Adds modules for alpha equivalence, name equivalence, and exact text matching. The module for exact text matching is complete, and some modifications to the matching algorithm are made in `matching-function.k` in order to incorporate it.
2. Also adds the functions `areNameEquivalent( )` and `areAlphaEquivalent( )` in the syntactic category `Fun` which will eventually check for name and alpha equivalence.
3. Changes some of the `imports` tags to fit in these modules appropriately.

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
Add link to corresponding JIRA issue.

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You assigned one person to review this PR

### If you are not a member of the core development team, please confirm:
- [ ] You signed the commit. Merging requires a signature. Please see the [developer wiki](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain) for instructions.
- [ ] Your GitHub account is also an account with [Travis CI](https://travis-ci.org). Unit tests will not run on your PR unless you have an account with Travis. Merge requires passed unit tests.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: rholang/src/main/k/rholang/name-equivalence.k [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module NAME-EQUIVALENCE
+imports FREE
+
+// A function that determines if two names are equivalent
+// This will be used to determine equivalence on a send/receive
+// Names are equivalent, modulo:
```
Comment (pyrocto) [inline rholang/src/main/k/rholang/name-equivalence.k]: Is `x!(0)` equivalent to `*@{x!(0)}`?

**Review by ghost — APPROVED** [review]

File: rholang/src/main/k/rholang/name-equivalence.k (hunk 1 above)
Comment (differentialderek) [reply rholang/src/main/k/rholang/name-equivalence.k]: In what context, as processes? e.g., are `@{ x!(0) }` and `@{ *@{ x!(0) }}` equivalent channels to send/receive on? I presume so. This module is meant to define a function that looks at names being sent/received on to see if they are equivalent.

Comment (differentialderek) [reply rholang/src/main/k/rholang/name-equivalence.k]: Certainly if `x` is bound, a send on the first will match with a receive on the second, so long as the pattern matches the process being sent.

Comment (differentialderek) [reply rholang/src/main/k/rholang/name-equivalence.k]: Ah, I see your point. I will add that to the list.

**Pull request merged**